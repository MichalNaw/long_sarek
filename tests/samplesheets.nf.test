nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_sarek"
    tag "cpu"

    test("-profile test --input tests/csv/3.0/sample_with_space.csv") {
        when {
            params {
                input = "${projectDir}/tests/csv/3.0/sample_with_space.csv"
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.failed
            assertAll(
                { assert snapshot(
                    workflow.stderr.toString().split(",").findAll { !it.contains("is available - Please consider updating your version to") && !it.contains("Validation of file failed") && !it.contains("Check script") }
                ).match() }
            )
        }
    }

    test("-profile test --step variant_calling --input tests/csv/3.0/recalibrated_somatic_two_normal_one_sample.csv") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
                input = "${projectDir}/tests/csv/3.0/recalibrated_somatic_two_normal_one_sample.csv"
                outdir = "$outputDir"
                step = "variant_calling"
            }
        }

        then {
            assert workflow.failed
            assertAll(
                { assert snapshot(
                    workflow.stderr.toString().split(",").findAll { !it.contains("is available - Please consider updating your version to") }
                ).match() }
            )
        }
    }
}
