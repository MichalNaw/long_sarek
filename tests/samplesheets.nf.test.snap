{
    "-profile test --step variant_calling --input tests/csv/3.0/recalibrated_somatic_two_normal_one_sample.csv": {
        "content": [
            [
                " Patient [test] has more than one sample [2] with normal status [0] and one sample with tumor status [1].]"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-13T21:38:02.778427213"
    },
    "-profile test --input tests/csv/3.0/sample_with_space.csv": {
        "content": [
            [
                " \u001b[0;31mThe following invalid input values have been detected:",
                " ",
                " \t-> Entry 2: Error for field 'sample' (test 2): \"test 2\" does not match regular expression [^\\S+$] (Sample ID must be provided",
                " cannot contain spaces and must be a string value)",
                " \u001b[0m"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-13T21:39:18.834066513"
    }
}